spec/dynamic/layout.md

```diff
@@ -0,0 +1,333 @@
+```rust
+#[repr(C)]
+pub union Repr{
+    pub V0: V0Repr,
+    pub V1: V1Repr,
+    ..pub Vn: VnRepr,
+}
+
+#[repr(C)]
+pub struct V0Repr(Int, V0F0, V0F1, ..V0Fm);
+#[repr(C)]
+pub struct V1Repr(Int, V1F0, V1F1, ..V1Fm);
+..#[repr(C)]
+pub struct VnRepr(Int, VnF0, VnF1, ..VnFm);
+```
+
+
+[§](r[dynamic.layout.enum.repr-c-int]) An `enum` definition with the `C,Int` repr-attribute, where `Int` is the name of an integer type, and with `n` variants `V0`, `V1`, `..Vn`, where variant `i<n` has `m` fields with types `ViF0`, `ViF1`, `..ViFm`, has an underlying type of the following *exposition only* `struct` declaration, which is modified by the `align(N)` repr attribute if the `enum` definition is
+```rust
+#[repr(C)]
+pub struct Repr(Int, VariantsRepr);
+#[repr(C)]
+pub union VariantsRepr{
+    pub V0: V0Repr,
+    pub V1: V1Repr,
+    ..pub Vn: VnRepr,
+}
+
+#[repr(C)]
+pub struct V0Repr(V0F0, V0F1, ..V0Fm);
+#[repr(C)]
+pub struct V1Repr(V1F0, V1F1, ..V1Fm);
+..#[repr(C)]
+pub struct VnRepr(VnF0, VnF1, ..VnFm);
+```
+
+[§](r[dynamic.layout.enum.repr-c]) An `enum` definition with the `C` repr-attribute is the same as the same definition with the `C,Int` repr-attribute, where `Int` is a *target dependenant* signed integer type.
+
+[§](r[dynamic.layout.enum.discrim]) For an enum declaration with the `C,Int` repr-attribute, or the `C` repr attribute, the first field of the exposition only `Repr` struct is initialized to the discriminant of the variant which is being represented, and the fields of the corresponding variant repr type are initialized to the fields of that variant. For an enum declaration with the `Int` repr-attribute, the first field of the variant repr type is the discriminant of the variant, and the remaining fields are initialized to the fields of that variant.
+
+[§](r[dynamic.layout.enum.size-align]) The size of an `enum` definition is at least sufficient to store its representation. The alignment of an `enum` definition is at least sufficient to satisfy the alignment requirement of all fields of all variants in its representation.
+
+[!NOTE]: In the case of an `enum` with a repr-attribute other than `repr(Rust)` or `repr(align(N))`, the size and alignment are given by the appropriate *exposition only* `Repr` type.
+
+[§](r[dynamic.layout.enum.repr-align]) If the `align(N)` repr-attribute is present on an `enum` definition, then the alignment requirement of the enum is at least `N`.
+
+[§](r[dynamic.layout.enum.option]) The special type `Option<T>` is an enum type, such that if `T` is one of the following types or a type with an underlying type that is one of the following (recursively), it has the corresponding underlying type.
+* `&T`: `*const T`
+* `&mut T`: `*mut T`
+* fn-ptr type: An unspecified, exposition-only, type with the same size and alignment as `fn()` that is valid for any initialized value
```

Comment:
I can't tell whether or not this accidentally restricts the value from carrying pointer parts. Add more words to clarify